### PR TITLE
Fix argument null exception when rendering an error page

### PR DESCRIPTION
**Pull request by kichalla**
@NTaylorMullen @Tratcher

### CODE REVIEW

**Comment by NTaylorMullen** [comment]
Question: Why was `frameContext` not being set?

File: src/Microsoft.AspNet.Diagnostics/Views/CompilationErrorPage.cs [hunk 1]
```diff
@@ -193,7 +193,7 @@ public CompilationErrorPage()
 #line hidden
 
 #line 48 "CompilationErrorPage.cshtml"
-                         if (frame.Line != 0 && frame.ContextCode.Any())
+                         if (frame.Line != 0 && frame.ContextCode !=null && frame.ContextCode.Any())
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNet.Diagnostics/Views/CompilationErrorPage.cs]: `frame.ContextCode?.Any() == true` for this and below

**Comment by kichalla** [comment]
@NTaylorMullen I am not exactly sure but we are populating this data for ContextCode
https://github.com/aspnet/Diagnostics/blob/dev/src/Microsoft.AspNet.Diagnostics/ErrorPageMiddleware.cs#L202

For this particular case it looks like below:
image

File: src/Microsoft.AspNet.Diagnostics/Views/CompilationErrorPage.cs (hunk 1 above)
Comment (kichalla) [inline src/Microsoft.AspNet.Diagnostics/Views/CompilationErrorPage.cs]: I think the one I have is probably more readable?

Comment (NTaylorMullen) [inline src/Microsoft.AspNet.Diagnostics/Views/CompilationErrorPage.cs]: Maybe it's just a repo preference, in Razor we do the `?. == true` trick a lot. Not vastly important.

**Comment by NTaylorMullen** [comment]
@kichalla Should definitely track down why it's being set to `null` (why it's failing to locate the responsible code). The original code has `null` checks for pre/post context code which makes me believe it was purposefully left without the normal context code `null` checks. 

Want to make sure we're not solving the wrong problem here.

**Comment by kichalla** [comment]
So on digging deeper I found that the real cause of this issue is with ErrorPage middleware not showing source code lines of Razor files. I filed a new issue here: https://github.com/aspnet/Diagnostics/issues/157

However the fix in this PR is still good as we want to avoid throwing argument null exception.

**Comment by NTaylorMullen** [comment]
Got it, so once #157 is completed the only way an error could occur is if a user put in a faulty `#line` pragma in their application (which would result in a `null` `frame.ContextCode`).

:shipit:

**Comment by kichalla** [comment]
commit 191ffe17219405c6c7c9607c20ba6dd6fbdeecde

**Pull request closed without merging**